google/resource_compute_network.go

```diff
@@ -6,12 +6,17 @@ import (
 
 	"github.com/hashicorp/terraform/helper/schema"
 	"google.golang.org/api/compute/v1"
+	"google.golang.org/api/googleapi"
+	"regexp"
 )
 
+const peerNetworkLinkRegex = "projects/(" + projectRegex + ")/global/networks/((?:[a-z](?:[-a-z0-9]*[a-z0-9])?))$"
```

Comment:
Personally, I'm OK with only accepting the first two forms. Terraform self links will always be using full or relative partial URLs, and while omitting the project is a nice shorthand, it's information that users will always have.

I'd be interested if we could file/bump an issue against Terraform itself for getting more information inside a `StateFunc`/`DiffSuppressFunc`; if we knew the `project` in either, we could allow the shorthand as well.